### PR TITLE
Add steps to localize the language of the app

### PR DESCRIPTION
**Pull request by ankitk20**
|PR checklist|Status|
|-----|:------:|
|Is issue created in GitHub?|[Yes](https://github.com/maqsoftware/Pehla-School/issues/271)|
|Is code analysis task executed and issues resolved in the Java/Kotlin code?|NA|
|Is code analysis task executed and issues resolved in the CPP/native code?|NA|
|Is the version code set in the AndroidManifest.xml or build.gradle file to 1 more than the version which has already been published?|NA|
|Is code commenting done properly for the changes made?|NA|
|Is correct variable naming convention followed?|NA|
|Is auto formatting shortcut Ctrl+Alt+Shift+L executed in all files?|NA|
|Are the message strings used in the application taken from strings.xml file?|NA|
|Are the permissions which are not required removed from AndroidManifest.java file?|NA|
|Is the _uses-feature_ tag used for only must have feature in AndroidManifest.java file?|NA|
|Are the generated files correctly configured in .gitignore?|NA|
|Are the release signing configurations removed from the gradle files?|NA|
|Is commented code removed from all files?|NA|
|Are updated images compressed (< 100 KB)?|NA|
|Ensured there are no typos in any string. For example: Variable/method names, hardcoded strings, commit messages, PR titles, issues, code comments?|Yes|

### CODE REVIEW

File: LOCALIZE.md [hunk 1]
```diff
@@ -1,67 +1,57 @@
...
+2. Add the new language in the `localeType` enum in LanguageManager.hpp file.
+3. Add a case statement in the `getLocalizedString()` method for a new language. Also, add a case statement for the new language in the files viz. `CompletePopup.cpp, TodoLoadingScene.cpp, NumberMatchingScene.cpp, MultipleChoicesScene.cpp, LetterMatchingScene.cpp, FillTheBlanksScene.cpp, LetterTraceMainDepot.cpp, ReorderingScene.cpp, CompTraceScene.cpp, MatchingScene.cpp, LRDividedTypeReorderingScene.cpp, LRDividedTypeFillTheBlanksScene.cpp, LRAllInOneTypeQuestionScene.cpp, LRDividedTypeQuestionScene.cpp, LRComprehensionScene.cpp, WordTraceMainDepot.cpp, AnswerPadMulti.cpp, AnswerPadSingle.cpp, GradeSelector.cpp.`
+4. Localize the [game tutorial](https://github.com/maqsoftware/Pehla-School-Assets/tree/master/localized/tutorialvideo) videos.
+4. [curriculumdata_levels_en.tsv](https://github.com/maqsoftware/Pehla-School-Assets/blob/master/localized/curriculumdata_en.tsv) file contains mapping of game names for different days. It contains titles like "PreSchool", "English 1" or "Math 3" that can be localized.
```
Comment (alankrutap01) [inline LOCALIZE.md]: 'contain' instructions

File: LOCALIZE.md [hunk 2]
```diff
@@ -1,67 +1,57 @@
...
+5. [eggquizliteracy_levels_en.tsv](https://github.com/maqsoftware/Pehla-School-Assets/blob/master/localized/games/eggquiz/eggquizliteracy_levels_en.tsv) and [eggquizmath_levels_en.tsv](https://github.com/maqsoftware/Pehla-School-Assets/blob/master/localized/games/eggquiz/eggquizmath_levels_en.tsv) files contains instructions for the questions to be answered in the quiz sections that can be localized.
+6. [wordwindow_levels_en.tsv](https://github.com/maqsoftware/Pehla-School-Assets/blob/master/localized/games/wordwindow/wordwindow_level_en.tsv) contains word problems for doing math that can be localized.
+7. Add locale specific splash screen and falling Pehla School logo images in [localized->system](https://github.com/maqsoftware/Pehla-School-Assets/tree/master/localized/system) and [main->system](https://github.com/maqsoftware/Pehla-School-Assets/tree/master/main/system) folders respectively.
+8. Localize all the resources in [res](https://github.com/maqsoftware/Pehla-School/tree/newmaster/pehlaschool/proj.android-studio/app/src/english/res) folder and create a new folder the new language.
```
Comment (alankrutap01) [inline LOCALIZE.md]: a new folder 'for' the new language

File: LOCALIZE.md [hunk 3]
```diff
@@ -1,67 +1,57 @@
...
+9. Add a new translation string to the vector in the CPP map initialization and a condition for the new language below it in the `DailyScene2`.cpp file.
```
Comment (alankrutap01) [inline LOCALIZE.md]: Add comma after 'below it'

File: LOCALIZE.md [hunk 4]
```diff
@@ -1,67 +1,57 @@
...
+10. Update the TTS locale settings according to the new language in VoiceMoldManager.java.
```
Comment (Shailin) [inline LOCALIZE.md]: Add a step to create language-specific AndroidManifest for Content Provider changes

File: LOCALIZE.md [hunk 5]
```diff
@@ -1,67 +1,57 @@
-# Localization steps for KitkitSchool Applications
+## Steps to localize only instructions
+1. Add a new `productFlavor` in the [build.gradle](https://github.com/maqsoftware/Pehla-School/blob/newmaster/pehlaschool/proj.android-studio/app/build.gradle) file along with `applicationSuffixId` property in it.
+2. Add the new language in the `localeType` enum in LanguageManager.hpp file.
+3. Add a case statement in the `getLocalizedString()` method for a new language. Also, add a case statement for the new language in the files viz. `CompletePopup.cpp, TodoLoadingScene.cpp, NumberMatchingScene.cpp, MultipleChoicesScene.cpp, LetterMatchingScene.cpp, FillTheBlanksScene.cpp, LetterTraceMainDepot.cpp, ReorderingScene.cpp, CompTraceScene.cpp, MatchingScene.cpp, LRDividedTypeReorderingScene.cpp, LRDividedTypeFillTheBlanksScene.cpp, LRAllInOneTypeQuestionScene.cpp, LRDividedTypeQuestionScene.cpp, LRComprehensionScene.cpp, WordTraceMainDepot.cpp, AnswerPadMulti.cpp, AnswerPadSingle.cpp, GradeSelector.cpp.`
```
Comment (Shailin) [inline LOCALIZE.md]: Add step to add language-specific titles and text in `LanguageManager.cpp`

File: LOCALIZE.md (hunk 4 above)
Comment (ankitk20) [reply LOCALIZE.md]: Not required

File: LOCALIZE.md [hunk 6]
```diff
@@ -1,67 +1,38 @@
...
-## Voice Engine (for English only) - Does not support localization
-Voice Engine is being used by "Word Note" activity in the Main App. This app does not support localization.  
+* #### Writing Board module
+  It uses images containing alphabets and words and a user is expected to trace them to learn to write.
+  1.  Update the images in [assets/image](https://github.com/maqsoftware/Pehla-School/tree/newmaster/pehlaschool/proj.android-studio/app/src/main/assets/image/en-us) folder with translated text in them.
+---
+### Pehla School Library
+It is a of a collection of storybooks and videos for reading and entertainment.
+1. Translate the content of [strings.xml](https://github.com/maqsoftware/Pehla-School/blob/newmaster/pehlaschoollibrary/proj.android-studio/app/res/values/strings.xml) file in the [Pehla School Library](https://github.com/maqsoftware/Pehla-School/tree/newmaster/pehlaschoollibrary/) project.
+2. [Library asset](https://github.com/maqsoftware/Pehla-School-Hindi-Assets-Library/tree/master/en-us) folder contains images and videos for learning and entertainment and images and videos which contain audio or text specific to a language should be updated.
```
Comment (alankrutap01) [inline LOCALIZE.md]: Add comma after entertainment and also after text.
Try to reframe the sentence as 'should be updated' part isn't clear.

File: LOCALIZE.md [hunk 7]
```diff
@@ -1,67 +1,38 @@
...
+2. Add the new language in the `localeType` enum in LanguageManager.hpp file.
+3. Add a key for the new language specific strings in LanguageManager.cpp file.
+4. Add a case statement in the `getLocalizedString()` method for a new language. Also, add a case statement for the new language in the files viz. `CompletePopup.cpp, TodoLoadingScene.cpp, NumberMatchingScene.cpp, MultipleChoicesScene.cpp, LetterMatchingScene.cpp, FillTheBlanksScene.cpp, LetterTraceMainDepot.cpp, ReorderingScene.cpp, CompTraceScene.cpp, MatchingScene.cpp, LRDividedTypeReorderingScene.cpp, LRDividedTypeFillTheBlanksScene.cpp, LRAllInOneTypeQuestionScene.cpp, LRDividedTypeQuestionScene.cpp, LRComprehensionScene.cpp, WordTraceMainDepot.cpp, AnswerPadMulti.cpp, AnswerPadSingle.cpp, GradeSelector.cpp.`
+5. Localize the [game tutorial](https://github.com/maqsoftware/Pehla-School-Assets/tree/master/localized/tutorialvideo) videos.
+6. [curriculumdata_levels_en.tsv](https://github.com/maqsoftware/Pehla-School-Assets/blob/master/localized/curriculumdata_en.tsv) file contain mapping of game names for different days. It contains titles like "PreSchool", "English 1" or "Math 3" that can be localized.
```
Comment (alankrutap01) [inline LOCALIZE.md]: 'contains'

File: LOCALIZE.md [hunk 8]
```diff
@@ -1,67 +1,38 @@
...
+7. [eggquizliteracy_levels_en.tsv](https://github.com/maqsoftware/Pehla-School-Assets/blob/master/localized/games/eggquiz/eggquizliteracy_levels_en.tsv) and [eggquizmath_levels_en.tsv](https://github.com/maqsoftware/Pehla-School-Assets/blob/master/localized/games/eggquiz/eggquizmath_levels_en.tsv) files contains instructions for the questions to be answered in the quiz sections that can be localized.
```
Comment (alankrutap01) [inline LOCALIZE.md]: 'contain'
Add comma after 'quiz sections'

File: LOCALIZE.md [hunk 9]
```diff
@@ -1,67 +1,38 @@
-# Localization steps for KitkitSchool Applications
+## Steps to localize only instructions
```
Comment (mathewnik90) [inline LOCALIZE.md]: Steps to localize instructions

File: LOCALIZE.md [hunk 10]
```diff
@@ -1,67 +1,38 @@
...
+2. Add the new language in the `localeType` enum in LanguageManager.hpp file.
+3. Add a key for the new language specific strings in LanguageManager.cpp file.
+4. Add a case statement in the `getLocalizedString()` method for a new language. Also, add a case statement for the new language in the files viz. `CompletePopup.cpp, TodoLoadingScene.cpp, NumberMatchingScene.cpp, MultipleChoicesScene.cpp, LetterMatchingScene.cpp, FillTheBlanksScene.cpp, LetterTraceMainDepot.cpp, ReorderingScene.cpp, CompTraceScene.cpp, MatchingScene.cpp, LRDividedTypeReorderingScene.cpp, LRDividedTypeFillTheBlanksScene.cpp, LRAllInOneTypeQuestionScene.cpp, LRDividedTypeQuestionScene.cpp, LRComprehensionScene.cpp, WordTraceMainDepot.cpp, AnswerPadMulti.cpp, AnswerPadSingle.cpp, GradeSelector.cpp.`
+5. Localize the [game tutorial](https://github.com/maqsoftware/Pehla-School-Assets/tree/master/localized/tutorialvideo) videos.
+6. [curriculumdata_levels_en.tsv](https://github.com/maqsoftware/Pehla-School-Assets/blob/master/localized/curriculumdata_en.tsv) file contains mapping of game names for different days. It contains titles like "PreSchool", "English 1" or "Math 3" that can be localized.
```
Comment (mathewnik90) [inline LOCALIZE.md]: No need to repeat that can be localized with each statement

File: LOCALIZE.md [hunk 11]
```diff
@@ -1,67 +1,38 @@
...
-Seaworld is an app where different fish appear on the screen. Localization is not needed for this application.  
-___
-## Voice Engine (for English only) - Does not support localization
-Voice Engine is being used by "Word Note" activity in the Main App. This app does not support localization.  
+* #### Writing Board module
+  It uses images containing alphabets and words and a user is expected to trace them to learn to write.
+  1.  Update the images in [assets/image](https://github.com/maqsoftware/Pehla-School/tree/newmaster/pehlaschool/proj.android-studio/app/src/main/assets/image/en-us) folder with translated text in them.
+---
+### Pehla School Library
+It is a of a collection of storybooks and videos for reading and entertainment.
```
Comment (mathewnik90) [inline LOCALIZE.md]: Grammar

File: LOCALIZE.md [hunk 12]
```diff
@@ -1,67 +1,38 @@
...
-## Lock Screen - Not required
-This app appears when the device wakes up. The device will be unlocked by touching the Kitkit School icon. Localization is not needed for this application.  
-___
-## Seaworld - Not required
-Seaworld is an app where different fish appear on the screen. Localization is not needed for this application.  
-___
-## Voice Engine (for English only) - Does not support localization
-Voice Engine is being used by "Word Note" activity in the Main App. This app does not support localization.  
+* #### Writing Board module
+  It uses images containing alphabets and words and a user is expected to trace them to learn to write.
```
Comment (mathewnik90) [inline LOCALIZE.md:30]: Why are the additional steps skipped?

Comment (ankitk20) [reply LOCALIZE.md:30]: They are not required anymore because simply updating the images not in the folder is sufficient.

**Review by mathewnik90 — APPROVED** [review]

**Pull request merged**